source/common/http/conn_manager_impl.cc

```diff
@@ -263,13 +263,23 @@ void ConnectionManagerImpl::doDeferredStreamDestroy(ActiveStream& stream) {
   if (connection_idle_timer_ && streams_.empty()) {
     connection_idle_timer_->enableTimer(config_.idleTimeout().value());
   }
+  if (requests_overflow_ && streams_.empty()) {
+    doConnectionClose(Network::ConnectionCloseType::FlushWrite, absl::nullopt, "");
```

Comment:
It seems like this shouldn't be needed. After you've called `startDrainSequence()`, the connection should get closed whenever all the streams are closed. See `checkForDeferredClose()` at the end of `doEndStream()`.

Reply:
Removed requests_overflow_ and above.